zerver/views/user_settings.py

```diff
@@ -357,6 +374,84 @@ def json_change_settings(
     # TODO: Change the cache flushing strategy to make sure cache
     # does not contain stale objects.
     user_profile = UserProfile.objects.get(id=user_profile.id)
+
+    # Loop over user_profile.property_types
+    request_settings = {
+        setting_name: requested_value
+        for setting_name, requested_value in locals().items()
+        if setting_name in user_profile.property_types and requested_value is not None
+    }
+
+    if target_users is not None:
+        if not user_profile.is_realm_admin:
+            raise JsonableError(
+                _("Only organization administrators can change settings for other users.")
+            )
+
+        if timezone is not None:
+            raise JsonableError(
+                _("Organization administrators cannot change timezone setting for other users.")
+            )
+
+        for setting_name in request_settings:
+            if (
+                setting_name in UserBaseSettings.SECURITY_SENSITIVE_USER_SETTINGS
+                or setting_name in ["email", "new_password"]
```

Comment:
Probably there should be a comment for these two fields noting that they're not functionally user settings, but still processed by this endpoint.

Reply:
Actually, is this code operative? Neither of those fields is inside `property_types`. 

Maybe should just be an `assert setting_name not in {"email", "new_password", "timezone"}`